### ISSUE TITLE
[Proposal] Add a feature to put Makefile in :code.priv_dir(:my_app)

### ISSUE THREAD

**Issue by zacky1972** [elixir-lang/elixir_make#47]
Hi, 

I proposed #41 but it was rejected because `Application.app_dir(:my_app, "somewhere")` doesn't work.

So, I propose an alternative approach that uses `:code.priv_dir(:my_app)`, again. I've proved it works by the following code:

```elixir
defmodule Mix.Tasks.MixTest do
  use Mix.Task
  require Logger

  def run(args) do
  	Logger.info(:code.priv_dir(:mix_test))
  end
end
```

**Comment by josevalim** [comment]
Application.app_dir is also going to work in your example above. That's because the code above runs after the project is compiled. As I replied on #41, :code.priv_dir won't solve the issue because Application.app_dir is implemented with `:code.priv_dir`, so the root issue was always :code.priv_dir.

**Comment by zacky1972** [comment]
I believe Makefile in `Application.app_dir` or `:code.priv_dir` works when #43. Pelemay uses this combination.

I propose that this feature will be available only when #43 is activated.

**Comment by zacky1972** [comment]
Of course, I'll contribute to send PR to fix this issue.
